src/test/scala/com/asidatascience/configuration/BaseSpec.scala

```diff
@@ -0,0 +1,43 @@
+package com.asidatascience.configuration
+
+import java.util.concurrent.atomic.AtomicInteger
+import scala.concurrent.duration._
+import scala.util.{Success, Try}
+
+import akka.actor.ActorSystem
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.{BeforeAndAfterAll, FlatSpec, Matchers}
+
+trait BaseSpec
+extends FlatSpec
```

Comment:
Interesting. I didn't know traits could extend a class. It looks like the benefits of doing this are smilar to the benefits of using a self type. Ie.

```scala
class FlatSpec

trait BaseSpec extends FlatSpec
```

is similar to:

```scala
class FlatSpec

trait BaseSpec {
  this: FlatSpec =>
}
```
inasmuch as you're forcing subclasses of `BaseSpec` to also be subclasses of `FlatSpec`.

Anyway, we can avoid all this complicated typing by just using an abstract class:

```scala
abstract class BaseSpec extends FlatSpec
```

This is definitely what scalatest recommends.